### PR TITLE
[KYUUBI #526] Support kyuubi.operation.scheduler.pool for queries

### PR DESCRIPTION
**Pull request by ulysses-you**
ulysses-you [Closes #528](https://github.com/yaooqinn/kyuubi/pull/528)    Test Plan Feature [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

Support scheduler pool which is native supported by Spark.

### _How was this patch tested?_
Pass CI
+
Manual tets

The default pool name is `default`, aflter execute `set kyuubi.operation.scheduler.pool = default1;`, the pool changed.

pool

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @pan3793 @turboFei @yaooqinn

**Review by pan3793 — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -533,4 +533,11 @@ object KyuubiConf {
     .version("1.2.0")
     .stringConf
     .createWithDefault("SHOW DATABASES")
+
+  val OPERATION_SCHEDULER_POOL: OptionalConfigEntry[String] = buildConf("operation.scheduler.pool")
+    .doc("The scheduler pool of job. Note that, this config should be used after change Spark " +
+      "config spark.scheduler.mode=FAIR.")
+    .version("1.1.1")
```
Comment (turboFei) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:540]: should be 1.2.0?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala [hunk 2]
```diff
@@ -41,6 +41,10 @@ class ExecuteStatement(
   private val forceCancel =
     session.sessionManager.getConf.get(KyuubiConf.OPERATION_FORCE_CANCEL)
 
+  private val schedulerPool =
+    spark.conf.getOption(KyuubiConf.OPERATION_SCHEDULER_POOL.key).orElse(
+      session.sessionManager.getConf.get(KyuubiConf.OPERATION_SCHEDULER_POOL))
```
Comment (turboFei) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:46]: should we fallback to spark.scheduler.pool?

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:46]: For ad-hoc use case,  I think kyuubi should be transparent for customer.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala (hunk 1 above)
Comment (ulysses-you) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:540]: Decide to make 1.1.1 support this minor feature for our requirement : )

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala (hunk 2 above)
Comment (ulysses-you) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:46]: `spark.scheduler.pool` just a local property that user can not set it even using `SparkThriftServer` so we don not need to respect it at config level.

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala:46]: make sense

**Review by turboFei — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks @pan3793 @turboFei  merging to master for v1.2.0. Seems here some conflicts with branch-1.1, let me do a backport in another PR.

**Comment by yaooqinn** [comment]
can have uts? it's not that hard to test this

**Pull request closed without merging**